### DISCUSSION TITLE
ExpressLRS Hardware Compatibility

### DISCUSSION THREAD

**Discussion by Junnicutt** [markqvist/Reticulum#580] [Ideas]
Would expresslrs hardware work with rnode? Obviously you'd need a USB to uart adapter but other than that is there any reason to think it wouldn't work? If so it could open a very large and cheap source of hardware and make it much easier to try out. I thought I was posting this in the rnode firmware repository but it keeps switching to reticulum so just close this if it's in the wrong location.

**Comment by cobraPA** [comment]
It looks like ExpressLRS has a lot of different hardware and is geared to RC control?  Most of the RNode firmware currently runs on hardware that uses a SemTech LoRa chip, so these are the easiest to make work.  Do you have a link to a particular ExpressLRS piece of hardware that shows the details of what radio chips are used?

**Reply by Junnicutt** [reply]
Yeah probably 90% use sx1280 or sx1276. For the mcu, cheaper older models use esp8266 but many of the diversity or higher end rx's use esp32. They also recently added support for esp32c3 and esp32s3 I think and also to a dual band 3rd gen chipset I can't remember right now but it's not common yet. It seems like a perfect fit for size and cost.  The pwm models even break out some extra pins. They have many versions up to 500mw for a postage stamp sized board. There's a few older versions that used stm32 but that's being fazed out. If you're looking for a good example anything made by matek is good quality.

**Reply by cobraPA** [reply]
Most esp32 are supported in other drivers, and so is sx1276. I think sx1280 was recently added as well, so it’s just a matter of adding models for the hardware you want to run.  (and testing it.)

**Comment by faragher** [comment]
I don't know about the hardware. However, your question about the USB adapter makes me think this information might be useful:

I have been running an RNode directly over UART serial for some time now. So long as the hardware can communicate over serial, you can talk to the RNode. Bear in mind the RNode is a modem, and doesn't do any Reticulum routing.

Does that at least partially answer your question?

**Reply by Junnicutt** [reply]
Thanks, Yeah that helps. I had just started checking out Rnode and reticulum and was curious wether the hardware I had already could work.
